### PR TITLE
enhancement(xref/ui): add autocomplete with fuzzy search

### PR DESCRIPTION
**Pull request by sidvishnoi**

### CODE REVIEW

File: static/xref/.prettierrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "singleQuote": true,
```
Comment (marcoscaceres) [inline static/xref/.prettierrc]: 🤢... just sayin

Comment (sidvishnoi) [reply static/xref/.prettierrc]: I know I know 😞 
There were mostly single quotes earlier in this file... ran prettier => too many changes.. hence why

File: static/xref/autocomplete.js [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+    option.setAttribute('role', 'option');
+    return option;
+  }
+};
+
+export function autocomplete(settings) {
+  const {
+    input,
+    debounceWaitMs = 0,
+    minLen = 1,
```
Comment (marcoscaceres) [inline static/xref/autocomplete.js]: didn't know this was a thing!

File: static/xref/index.html [hunk 3]
```diff
@@ -9,24 +9,24 @@
       try{customElements.define('built-in',document.createElement('p').constructor,{'extends':'p'})}
       catch(e){document.write('<script src="//unpkg.com/@ungap/custom-elements-builtin"><\x2fscript>')}
     </script>
-    <script src="script.js" defer></script>
+    <script src="fuse.js?v3.4.5"></script>
+    <script type="module" src="script.js" defer></script>
```
Comment (sidvishnoi) [inline static/xref/index.html]: I think I'll add a version here too and make the js/css cache immutable

File: static/xref/script.js [hunk 4]
```diff
@@ -1,57 +1,57 @@
...
-      options.add(value);
-      const option = document.createElement('option');
-      option.value = value;
-      datalist.appendChild(option);
+  connectedCallback() {
+    this.fuse = new Fuse(this.options, {});
+    const limit = parseInt(this.dataset.limit, 10) || 10;
+    autocomplete({
+      input: this,
+      fetch: (text, update) => {
```
Comment (marcoscaceres) [inline static/xref/script.js]: using fat arrow here makes this ambiguous...

File: static/xref/script.js [hunk 5]
```diff
@@ -1,57 +1,57 @@
...
+        text = text.toLowerCase();
+        const matchedIndexes = this.fuse.search(text).slice(0, limit);
+        const values = matchedIndexes.map(i => this.options[i]);
+        update(values);
+      },
+      onSelect: value => {
```
Comment (marcoscaceres) [inline static/xref/script.js]: same here... can this just be a function on the object?

File: static/xref/script.js (hunk 4 above)
Comment (sidvishnoi) [reply static/xref/script.js]: `this` refers to current input element. will make it clear with an assignment and no fat arrow..

File: static/xref/script.js [hunk 6]
```diff
@@ -80,10 +80,10 @@ const options = {
 };
 
 function getFormData() {
-  const term = form.querySelector("input[name='term']").value;
-  const specs = form.querySelector("input[name='cite']").values;
-  const types = form.querySelector("input[name='types']").values;
-  const forContext = form.querySelector("input[name='for']").value;
+  const term = form.term.value;
```
Comment (marcoscaceres) [inline static/xref/script.js]: Maybe make the aliasing more obvious?

```suggestion
  const {value: term} = form.term;
```

File: static/xref/script.js [hunk 7]
```diff
@@ -146,7 +148,15 @@ function howToCiteIDL(term, entry) {
   switch (type) {
     case 'exception':
       // Except for the following, exceptions take the form {{"SomeException"}}
-      if (!['EvalError', 'RangeError', 'ReferenceError', 'TypeError', 'URIError'].includes(term)) {
+      if (
+        ![
```
Comment (marcoscaceres) [inline static/xref/script.js]: Maybe we should make this a const above or outside the function entirely? Less ugly.

**Pull request merged**